### ISSUE TITLE
Support `<script setup>` and SFC css var injection for Vue 2.7.0+ in `no-unsupported-features` rule

### ISSUE THREAD

**Issue by danielwaltz** [vuejs/eslint-plugin-vue#1916]
**What rule do you want to change?**
`no-unsupported-features`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings, as `<script setup>` and CSS var injection are both supported features in Vue `>=2.7.0`

**How will the change be implemented? (New option, new default behavior, etc.)?**
I plan to update the rule to include `<script setup>` and CSS var injection as part of the `>=2.7.0` version.

**Please provide some example code that this change will affect:**

```vue
<script setup>
</script>
```

**What does the rule currently do for this code?**
Displays an error/warning with a `{ version: '2.7.0' }` rule option set

**What will the rule do after it's changed?**
Will pass with a `{ version: '2.7.0' }` rule option set

**Additional context**
